### PR TITLE
Add mruby-string-bitops gem

### PR DESCRIPTION
**Pull request by hasumikin**
Implement String bit operations from CRuby
(spec in https://bugs.ruby-lang.org/issues/22118): `bit_get`, `bit_set?`, `bit_set`, `bit_clear`, `bit_flip`, `bit_count`, and `bitwise_not/and/or/xor` with their bang`!` variants.

The port minds mruby-specific concerns: the bulk kernels use pointer-width words so 32-bit targets avoid emulated 64-bit arithmetic, word-aligned buffers get true word loads even on cores without unaligned access support (e.g. Cortex-M0+), and bit offsets are `mrb_int` with no Bignum path. See the gem's README.md for details and the intentional differences from CRuby.

I didn't add this gem to any gembox. @matz I leave this matter to you

## Summary by CodeRabbit

* **New Features**
  * Added `mruby-string-bitops` with String bit access, mutation, counting, inversion, and binary bitwise operations.
  * Supports configurable bit order, operand conversion, large strings, and both in-place and non-mutating variants.
  * Added validation for invalid offsets, mismatched lengths, frozen strings, and unsupported inputs.

* **Documentation**
  * Added comprehensive usage guidance, examples, return values, errors, encoding behavior, and compatibility notes.

* **Tests**
  * Added coverage for bit operations, coercion, edge cases, encodings, and error handling.

### CODE REVIEW

File: mrbgems/mruby-string-bitops/src/string_bitops.c [hunk 1]
```diff
@@ -0,0 +1,587 @@
...
+{
+  return bitop_mutate(mrb, str, BITOP_MUT_FLIP);
+}
+
+static mrb_value
+mrb_str_bit_count(mrb_state *mrb, mrb_value str)
+{
+  mrb_get_args(mrb, "");
+  return mrb_int_value(mrb, bitop_count_bits((const unsigned char*)RSTRING_PTR(str), RSTRING_LEN(str)));
+}
```
[the remark this answers is not in the document]
Comment (hasumikin) [reply mrbgems/mruby-string-bitops/src/string_bitops.c:436]: Fixed in a3433ef

File: mrbgems/mruby-string-bitops/src/string_bitops.c [hunk 2]
```diff
@@ -0,0 +1,587 @@
...
+  case MRB_TT_COMPLEX:
+    break;
+  default:
+    if (mrb_respond_to(mrb, index, MRB_SYM(to_int))) {
+      mrb_value converted = mrb_funcall_argv(mrb, index, MRB_SYM(to_int), 0, NULL);
+      if (mrb_type(converted) == MRB_TT_INTEGER || mrb_type(converted) == MRB_TT_BIGINT) {
+        index = converted;
+        break;
+      }
+    }
```
[the remark this answers is not in the document]
Comment (hasumikin) [reply mrbgems/mruby-string-bitops/src/string_bitops.c:290]: I don't think we need to call mrb_exc_raise. mrb_funcall_argv should propagate an exception by longjmp, right?

File: mrbgems/mruby-string-bitops/test/string_bitops.rb [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+  # A to_int returning a value beyond mrb_int raises RangeError, the
+  # same as passing such a value directly.  Without mruby-bigint the
+  # power itself raises RangeError; with it, the Bigint reaches the
+  # offset conversion.  Both configurations end in RangeError.
+  big = Object.new
+  def big.to_int
+    2 ** 100
+  end
+  assert_raise(RangeError) { s.bit_get(big) }
```
[the remark this answers is not in the document]
Comment (hasumikin) [reply mrbgems/mruby-string-bitops/test/string_bitops.rb:47]: Wrong. Integer overflow in mrb_int_pow calls mrb_int_overflow(mrb, "power"), which raises RangeError
